chain-extensions/pallet-assets/src/lib.rs

```diff
@@ -364,6 +432,144 @@ where
                 let decimals = pallet_assets::Pallet::<T>::decimals(&id);
                 env.write(&decimals.encode(), false, None)?;
             }
+            AssetsFunc::StartDestroy => {
```

Comment:
My recommendation is to remove all `destroy` related functions from the interface, because of the reasons already discussed before. This might change again in the future, leaving us with potentially incompatible chain-extension.

I'd suggest to document that in case of needing to destroy the asset, `transfer_ownership` can be used, after which required `destroy` functions can be called by an externally owned account.